### PR TITLE
Add `Series` copy binding

### PR DESCRIPTION
**Pull request by love-lena**
Add binding for a deep copy of a Column

---------
Backed by a deep-copy constructor in cudf::column, so functionally very little to go wrong.
All cudf tests pass.

### CODE REVIEW

File: modules/cudf/src/node_cudf/column.hpp [hunk 1]
```diff
@@ -653,6 +653,9 @@ struct Column : public EnvLocalObjectWrap<Column> {
     cudf::out_of_bounds_policy bounds_policy = cudf::out_of_bounds_policy::DONT_CHECK,
     rmm::mr::device_memory_resource* mr      = rmm::mr::get_current_device_resource()) const;
 
+  Column::wrapper_t copy(
+    rmm::mr::device_memory_resource* mr = rmm::mr::get_current_device_resource()) const;
+
```
Comment (trxcllnt) [inline modules/cudf/src/node_cudf/column.hpp:658]:
```suggestion
```

**Review by trxcllnt — CHANGES_REQUESTED** [review]
Looks great! Just one suggestion to remove the `Column::copy()` overload that takes `memory_resource*`. Declarations in headers without a corresponding definition in a `.cpp` file is how we get those symbol lookup errors at runtime.

File: modules/cudf/src/series.ts [hunk 2]
```diff
@@ -650,6 +650,25 @@ export class AbstractSeries<T extends DataType = any> {
     return this.__construct(this._col.gather(selection._col));
   }
 
+  /**
+   * Return a copy of this Series.
+   *
+   *
```
Comment (mzegar) [inline modules/cudf/src/series.ts:656]:
```suggestion
   *
```

Extremely nit picky, but I think we leave one `*` space inbetween.

File: modules/cudf/src/series.ts [hunk 3]
```diff
@@ -650,6 +650,25 @@ export class AbstractSeries<T extends DataType = any> {
...
+   * const a = Series.new(["foo", "bar", "test"]);
+   *
+   * a.copy() // StringSeries ["foo", "bar", "test"]
+   * ```
+   */
+  copy(memoryResource?: MemoryResource): Series<T> {
+    // const lhs = <Column>(this._col);
+    return Series.new(this._col.copy(memoryResource));
+  }
+  // this.__construct(this._col.copy()); }
```
Comment (mzegar) [inline modules/cudf/src/series.ts:670]: Are these comments needed?
`// const lhs = <Column>(this._col);` - line 667

`// this.__construct(this._col.copy()); }` - line 670

**Review by mzegar — COMMENTED** [review]
🎉 First binding down!

File: modules/cudf/src/node_cudf/column.hpp [hunk 4]
```diff
@@ -653,6 +653,9 @@ struct Column : public EnvLocalObjectWrap<Column> {
     cudf::out_of_bounds_policy bounds_policy = cudf::out_of_bounds_policy::DONT_CHECK,
     rmm::mr::device_memory_resource* mr      = rmm::mr::get_current_device_resource()) const;
 
+  Column::wrapper_t copy(
+    rmm::mr::device_memory_resource* mr = rmm::mr::get_current_device_resource()) const;
```
Comment (mzegar) [inline modules/cudf/src/node_cudf/column.hpp:657]: I could be wrong, but doesn't this need to be defined inside of `copying.cpp`?

```cpp
Column::wrapper_t Column::copy(
  rmm::mr::device_memory_resource* mr = rmm::mr::get_current_device_resource()) const {
    ...
}
```

File: modules/cudf/src/series.ts (hunk 2 above)
Comment (love-lena) [reply modules/cudf/src/series.ts]: love a good nit pick, resolved

**Review by trxcllnt — APPROVED** [review]

**Pull request merged**